compiler/src/dotty/tools/dotc/printing/DecompilerPrinter.scala

```diff
@@ -15,9 +16,15 @@ class DecompilerPrinter(_ctx: Context) extends RefinedPrinter(_ctx) {
   override protected def filterModTextAnnots(annots: List[untpd.Tree]): List[untpd.Tree] =
     annots.filter(_.tpe != defn.SourceFileAnnotType)
 
-  override protected def blockText[T >: Untyped](trees: List[Trees.Tree[T]]): Text = {
-    super.blockText(trees.filterNot(_.isInstanceOf[Closure[_]]))
-  }
+  override protected def blockToText[T >: Untyped](block: Block[T]): Text =
+    block match {
+      case Block((meth @ DefDef(nme.ANON_FUN, _, _, _, _)) :: Nil, _: Closure[T]) =>
```

Comment:
Extract the RHS from the `DefDef`(..., rhs)` directly.

Reply:
Should we extract the RHS using `unapply`? We have something like this:
```scala
case class DefDef(..., private var preRhs: LazyTree) {
  def rhs(implicit ctx) = forceIfLazy
}
```
I expect that the RHS should be accessed through `def rhs`

Reply:
Right